osu.Game/Users/UserActivity.cs

```diff
@@ -41,6 +41,12 @@ public abstract class UserActivity
 
         public virtual Color4 GetAppropriateColour(OsuColour colours) => colours.GreenDarker;
 
+        /// <summary>
+        /// Returns the ID of the beatmap involved in this activity, if applicable and/or available.
+        /// </summary>
+        /// <param name="hideIdentifiableInformation"></param>
+        public virtual int? GetBeatmapID(bool hideIdentifiableInformation = false) => null;
```

Comment:
One may say this should not exist in `UserActivity` itself, but the only other option I can think of is to define an interface that exposes this method and implement such interface on `InGame` and `EditingBeatmap`, and I'm smelling some scent of over-engineering in that so I just chose to define it at a base level for now.